### PR TITLE
Feature/uar 1101 retrieve trust data in journey

### PR DESCRIPTION
**Pull request by AndreKramer7654**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-1101

https://companieshouse.atlassian.net/browse/UAR-109 Epic
https://companieshouse.atlassian.net/browse/UAR-1101 Get and map Trust data
https://companieshouse.atlassian.net/browse/UAR-1187 Get and map Beneficial Owner and Trust links
https://companieshouse.atlassian.net/browse/UAR-1189 Get and map Individual and former individual Trustees
[UAR-1190](https://companieshouse.atlassian.net/browse/UAR-1190) Get and map Legal Entity and former legal entity trustee data

extends:

https://companieshouse.atlassian.net/browse/UAR-224 Retrieve Trusts private data to UI
https://companieshouse.atlassian.net/browse/UAR-1145 Get all trusts for an OE from ROE API
https://companieshouse.atlassian.net/browse/UAR-1142 Get all BO links to trusts for an OE from ROE API
https://companieshouse.atlassian.net/browse/UAR-1146 Get all Individual Trustees for an OE from ROE API
https://companieshouse.atlassian.net/browse/UAR-1147 Get all Corporate Trustees for an OE from ROE API

Change description
Fetch and map trust data into Web App.
Using https://github.com/companieshouse/api-sdk-node/pull/620

retrieve trust data in controllers for change and no-change if managed trust feature flag true

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

[UAR-1190]: https://companieshouse.atlassian.net/browse/UAR-1190?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/config/index.ts [hunk 1]
```diff
@@ -49,6 +49,7 @@ export const FEATURE_FLAG_ENABLE_ROE_REMOVE = getEnvironmentValue('FEATURE_FLAG_
 export const FEATURE_FLAG_DISABLE_UPDATE_PRIVATE_DATA_FETCH = getEnvironmentValue('FEATURE_FLAG_DISABLE_UPDATE_PRIVATE_DATA_FETCH_28062023', 'false');
 export const FEATURE_FLAG_ENABLE_UPDATE_STATEMENT_VALIDATION = getEnvironmentValue('FEATURE_FLAG_ENABLE_UPDATE_STATEMENT_VALIDATION_05072023', 'false');
 export const FEATURE_FLAG_ENABLE_UPDATE_MANAGE_TRUSTS = getEnvironmentValue('FEATURE_FLAG_ENABLE_UPDATE_MANAGE_TRUSTS_29082023', 'false');
+export const FEATURE_FLAG_DISABLE_UPDATE_TRUST_DATA_FETCH = getEnvironmentValue('FEATURE_FLAG_DISABLE_UPDATE_TRUST_DATA_FETCH_06102023', 'true');
```
Comment (AndreKramer7654) [inline src/config/index.ts:52]: temporary flag to disable the trustees and links for now so we can focus on just trust data first,

File: src/utils/update/trust.model.fetch.ts [hunk 2]
```diff
@@ -97,10 +97,10 @@ const fetchAndMapIndivdualTrustees = async (
   overseasEntityId: string,
   trust: Trust
 ) => {
-  const individualTrustees = await getIndividualTrustees(req, transactionId, overseasEntityId, trust.trust_id);
+  const individualTrustees = await getIndividualTrustees(req, transactionId, overseasEntityId, trust.ch_reference ?? trust.trust_id);
```
Comment (bbell-companieshouse) [inline src/utils/update/trust.model.fetch.ts:100]: Is the trust id needed here at all? (And in the other appearances in this file)

Comment (AndreKramer7654) [reply src/utils/update/trust.model.fetch.ts:100]: It's a fall back ch_reference should never be undefined but maybe better to be more careful about that.

Comment (AndreKramer7654) [reply src/utils/update/trust.model.fetch.ts:100]: I'll address on https://github.com/companieshouse/overseas-entities-web/pull/1146

**Comment by AndreKramer7654** [comment]
superseded by https://github.com/companieshouse/overseas-entities-web/pull/1146

**Pull request closed without merging**